webxrlayers-1.bs

```diff
@@ -797,6 +800,69 @@ interface XRMediaBinding {
 
 </pre>
 
+WebXR Device API Integration {#webxrintegration}
+============================
+
+XRRenderState changes {#xrrenderstatechanges}
+---------------------
+This module extends the {{XRRenderStateInit}} and {{XRRenderState}} interfaces with a new optional array {{XRRenderStateInit/layers}} containing
+instances of {{XRLayer}} or {{XRWebGLLayer}}.
+
+<pre class="idl">
+typedef (XRWebGLLayer or XRLayer) XRGenericLayer;
+
+dictionary XRRenderStateInit {
+  double depthNear;
+  double depthFar;
+  double inlineVerticalFieldOfView;
+  XRWebGLLayer? baseLayer;
+  FrozenArray&lt;XRGenericLayer&gt;? layers;
+};
+
+[SecureContext, Exposed=Window] interface XRRenderState {
+  readonly attribute double depthNear;
+  readonly attribute double depthFar;
+  readonly attribute double? inlineVerticalFieldOfView;
+  readonly attribute XRWebGLLayer? baseLayer;
+  readonly attribute FrozenArray&lt;XRGenericLayer&gt;? layers;
+};
+</pre>
+
+The <dfn dict-member for="XRRenderStateInit">layers</dfn> is an array containing
+instances of {{XRLayer}} or {{XRWebGLLayer}} to hand to the [=XR Compositor=].
+
+The <dfn attribute for="XRRenderState">layers</dfn> attribute returns an array containing
+the instances of {{XRLayer}} or {{XRWebGLLayer}} that are displayed by the [=XR Compositor=].
+
+<div class="algorithm" data-algorithm="initialize-renderstate-for-layers">
+When an {{XRRenderState}} object |state| is created for an {{XRSession}} |session| with {{layers}} enabled, the user agent MUST <dfn dfn-for="layers">initialize the render state</dfn> by running the following steps:
+  1. Initialize |state| by running the steps to [=initialize the render state=].
+  1. Initialize |state|'s {{XRRenderState/layers}} to an empty array.
+
+</div>
+
+updateRenderState changes {#updaterenderstatechanges}
+-------------------------
+<div class="algorithm" data-algorithm="updateRenderState-for-layers">
+When the <dfn method for="XRSession" dfn-for="layers">updateRenderState(|newState|)</dfn> method is called on an {{XRSession}} |session|
+with {{layers}} enabled, the user agent MUST run the following steps:
+
+  1. Run {{XRSession/updateRenderState}} with |newState|.
+  1. If |newState|'s {{XRRenderStateInit/baseLayer}} is set, set |session|'s [=pending render state=] to <code>null</code>, throw an {{InvalidStateError}} and abort these steps.
```

Comment:
Ah, I didn't remember where we had landed on this. Just to double check: The intent is that if we enabled "layers" in the optional/requiredFeatures then `baseLayer` is disabled entirely, right? I think that's fine, but we'll need to ensure pages have a way to detect that layers did actually get enabled in the case that they passed it as an optional feature, so that the developers know which render state to set.

Reply:
> Ah, I didn't remember where we had landed on this. 

Yes, I think we discussed it but it's not clear from the readme what we had decided.

> Just to double check: The intent is that if we enabled "layers" in the optional/requiredFeatures then `baseLayer` is disabled entirely, right? 

Correct

> I think that's fine, but we'll need to ensure pages have a way to detect that layers did actually get enabled in the case that they passed it as an optional feature, so that the developers know which render state to set.

The way I tried to structure the spec is that you will get an empty layers array in the XRRenderState (as opposed to it being absent).

Reply:
> The way I tried to structure the spec is that you will get an empty layers array in the XRRenderState (as opposed to it being absent).

Excellent, thanks!